### ISSUE TITLE
WKT2 / PROJJSON

### ISSUE THREAD

**Issue by m-mohr** [DanielJDufour/spatial-reference-system#4]
Any plans to support WKT2 and/or PROJJSON?

**Comment by DanielJDufour** [comment]
most definitely within the scope/vision of this project!

A few questions:
- would you like to work on this?
- what's the level of urgency/priority?  (just trying to balance against getting a new version of georaster layer getting out)

**Comment by m-mohr** [comment]
No, no plans right now. I don't even know how to convert between WKT and WKT2. This was more a question whether you have any ideas or libraries at hand.
pro4js doesn't support WKT2 / PROJJSON, but STAC does only expose this (+ EPSG), so there's a tech mismatch in the JS ecosystem for STAC + modern PROJ and I still hope for a solution ;-)

**Comment by DanielJDufour** [comment]
The solution I've used to convert between formats has been to just scrape epsg.io or gdalsrsinfo for all the different variants of a projection.  An example of this is seen here: https://github.com/DanielJDufour/proj-data.  I heard at FOSS4G that I should be using projinfo, though, so I'll probably switch to that.  Using https://github.com/danieljdufour/get-epsg-code to grab the epsg code, we could then just lookup the row in proj-data for the desired format (proj4, esriwkt, wkt, wkt2, etc.).

Because wkt2 and projjson are interoperable, I think we could store wkt2 in proj-data and convert to projjson from wkt2 on the fly when needed.  I think creating a new repo/project called projjson that would convert between the two would be achievable.  [wkt-crs](https://github.com/DanielJDufour/wkt-crs) can easily parse wkt2 into JavaScript objects/arrays, so I think we'd just need to create some code to reformat those into projjson.

If this sounds like a good idea to you, I'm wondering where such a repo should live?  We could create the repo under the stac-utils org and then talk to folks about maybe migrating it to https://github.com/proj4js once it's reached 1.0.0.  Or I could create it under https://github.com/geotiff.

Thoughts?

**Comment by m-mohr** [comment]
Ah, so this is for a "pre-defined" list of CRS'es only? I thought more of a general way to "convert" between WKT2 and WKT without a "database". But I'm not actually sure whether that's feasible at all (as WKT2 is more "concrete"/detailed than WKT). The conversion from WKT2 to PROJJSON and vice-verse is probably relatively straightforward, indeed.

I guess we can start in a personal repo, but I could imagine that Andreas is open to also host it in the proj4js org at some point. See also https://github.com/proj4js/proj4js/issues/400

Edit: But maybe it's not even a good idea https://github.com/rouault/projjson_to_wkt/issues/1

**Comment by DanielJDufour** [comment]
Hi, didn't want to leave this hanging.  All good points you are making.  You are correct, that the solution that I've been proposing would only work for a pre-defined list of CRS (those with EPSG codes).
